compiler/AST/FnSymbol.cpp

```diff
@@ -82,14 +83,9 @@ FnSymbol::FnSymbol(const char* initName)
 FnSymbol::~FnSymbol() {
   cleanupIteratorInfo(this);
   cleanupIteratorGroup(this);
-
+  cleanupCacheInfo(this);
   BasicBlock::clear(this);
-  delete basicBlocks;
-
-  if (calledBy) {
-    delete calledBy;
-    calledBy = NULL;
-  }
```

Comment:
Why does this remove `delete basicBlocks` and the conditional?

Reply:
* `delete basicBlocks` is done as part of `asicBlock::clear(this)`
* the conditional is unnecessary because `delete calledBy` performs the same checking and does nothing when `calledBy` is already null
* setting `calledBy` to null is also unnecessary because it is in the destructor and because we do not set deleted fields to null in some other destructor cases

So overall these changes lead to shorter code that does the same work.